interp/type.go

```diff
@@ -935,6 +935,17 @@ func isInterface(t *itype) bool {
 
 func isStruct(t *itype) bool { return t.TypeOf().Kind() == reflect.Struct }
 
+func isBool(t *itype) bool {
+	switch t.TypeOf().Kind() {
+	case reflect.Bool:
+		return true
+	case reflect.Chan:
+		return t.TypeOf().Elem().Kind() == reflect.Bool
```

Comment:
Can you explain this part?

Reply:
This is necessary to handle `if` expression on a boolean channel, like:
```go
channel := make(chan bool)
if <-channel {
 ....
}
```

Reply:
Odd. `<-(chan bool)` is `bool`; `chan bool` is `chan bool`. I would love to understand how this is all constructed here.

Reply:
Good question! the type computation of index expression resolving to a bool and function call returning a bool, as performed during CFG in  `nodeType()`,  returns a `bool` type as expected. So aanalysis of a  chan recv expression on a boolean channel should produce the same. I need to  take a closer look, but it may be because i do not take the channel direction in account when performing type analysis on channel expression.

Reply:
The problem was that type from unary operator `<-` was the channel type instead of the channel data type itself. Fixing that removes the need for special processing in `isBool()` , allows other simplifications and probably prevents further bugs. Thanks again for pointing that!